users/serializers.py

```diff
@@ -15,11 +15,33 @@
 
 
 class DataOceanUserSerializer(serializers.ModelSerializer):
+
+    def validate(self, data):
+        """
+        validate fop and legal_entity fields
+        """
+        if data.get('persone_status') == 'fop' and 'legal_entity':
+            if data.get('iban') == '':
+              raise serializers.ValidationError(_("enter iban please"))
```

Comment:
you missed field name 
```suggestion
              raise serializers.ValidationError({'iban': _("This field is required for company or FOP"}))
```